### PR TITLE
globalbranding story

### PR DESCRIPTION
**Pull request by baptisten**
Let me know any update needed

### CODE REVIEW

File: src/css/scheme-default.css [hunk 1]
```diff
@@ -26,6 +26,7 @@
   /* Fonts */
   --font-primary: 'Inter', sans-serif;
   --font-secondary: 'Lato', sans-serif;
+  --font-global-brand: 'Lora' ,'Georgia' ,'Cambria' ,'Times New Roman','Times' ,serif;
```
Comment (sleepingmonk) [inline src/css/scheme-default.css:29]: Should probably add Lora font to the @import at the top of the file:

https://fonts.google.com/specimen/Lora?query=lora

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.global-branding a {
+    font-family: --font-global-brand;
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.css:2]: Should be `var(--font-global-brand)`.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    font-size: var(--s0);
+    padding: 0 var(--s0);
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.css:4]: I'll pull this down and view the results but I'm suspicious of this padding on the `a`.  

My thought would be that any space between this text and other elements would be managed by the context that wraps them.  

For example, the Hat will be a cluster, which will include a gap between this and the secondary nav component automatically.

```
<div class="hat">
  <div class="container box"> <!-- Container for max width, box for edge padding. -->
      <div class="cluster"> <!-- Cluster manages relationship/space between child elements. -->
         <GlobalBranding> <!-- Padding on this would misalign it with the left edge of all other content. -->
         <SecondaryNav>
      </div>
   </div>
</div>
```

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    color: var(--color-white);
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.css:5]: Set this to: 
`color: var(--color, var(--color-white));`

This way it can inherit the color it needs from the parent element, or fall back to white.

From the hat example:
```
<div class="hat" style="--color: var(--color-white);">
```

Then this sub component, as well as the secondary nav link color can be set by the context (hat) which is what will determine its background color.  This way we can avoid having to override this low level color value if we find our selves in a sub footer or some other context that has a light background.

In other words, the lower level the element, the less assumptions it should make about its style properties.

This is how we avoid overrides and make the most out of the natural cascade of CSS.  It eliminates SOOOO much code.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.data.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    variant: "default",
+    text: "Judicial Branch of California",
+    destination: "https://courts.ca.gov",
+    sub_component_data: {
+      variant: "default",
+      attributes: new drupalAttribute(),
+    },
+    attributes: new drupalAttribute()
+  },
+  another_variant: {
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.data.js:15]: Delete the example variant.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    text-transform: uppercase;
+    text-decoration: none;
+}
+.global-branding a:hover {
+    text-decoration: underline;
+}
+
+
+.global-branding.global-branding--inverted a{
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.css:14]: See color comments above.  This can be removed if the context sets the color.  We don't need an "inverted" variant.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.data.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+// To mock drupal attributes in storybook.
+import drupalAttribute from 'drupal-attribute'
+
+export default {
+  default: {
+    variant: "default",
+    text: "Judicial Branch of California",
+    destination: "https://courts.ca.gov",
+    sub_component_data: {
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.data.js:9]: There's no sub component, so this can be removed.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.stories.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import readme from './readme.md';
+import data from './GlobalBranding.data';
+import GlobalBranding from './GlobalBranding.twig';
+import './GlobalBranding.css';
+
+export default {
+  title: 'Atoms/GlobalBranding',
+  component: GlobalBranding,
+  parameters: {
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.stories.js:9]: You can add a Storybook parameter here just for presentational aesthetics.
`layout: "centered",`

If that does something weird, like make the text wrap, try "padded" instead.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.stories.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    notes: readme,
+  },
+  argTypes: {
+    // Don't show the mock drupal attributes.
+    attributes: {table: {disable: true}},
+    // Variant select is a pretty common arg.
+    variant: {
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.stories.js:16]: No variant, so feel free to remove.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.twig [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+{# Set up variant classes. #}
+{% set variant_class = variant ? 'global-branding--' ~ variant : 'global-branding--default' %}
+
+{% set classes = [
+  'global-branding',
+  variant_class,
+] %}
+
+<div {{ attributes.addClass(classes) }}>
+  <div class="global-branding__wrapper">
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/GlobalBranding.twig:17]: We probably don't need these additional __wrapper and __inner divs.  They're not used for styling anything.

File: src/stories/Atoms/GlobalBranding/readme.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+<!-- This is the general documentation layout. Add or remove any sections as needed, but try to stay consistent across components. -->
+# Global Branding
+
+Description of the Global Branding in markdown.
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/readme.md:4]: Update this to describe what this component is, does, or is for.

File: src/stories/Atoms/GlobalBranding/readme.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+<details>
+  <summary>Inherited CSS Variables:</summary>
+  - `--name`: description...
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/readme.md:8]: `--color`: Sets the font color.

File: src/stories/Atoms/GlobalBranding/readme.md [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+</details>
+
+<details>
+  <summary>Twig Variables:</summary>
+  ```
+  variant: "default",
```
Comment (sleepingmonk) [inline src/stories/Atoms/GlobalBranding/readme.md:14]: Once you've finalized the "data.js" for this, copy the structure here for easy reference by anyone integrating this, using the docs as a guide.

**Review by sleepingmonk — CHANGES_REQUESTED** [review]
Thanks @baptisten This is pretty good.  Just a couple of notes.

Seems like a lot of comments but they're really small touch ups and additional guidance for context.

File: src/css/scheme-default.css (hunk 1 above)
Comment (baptisten) [reply src/css/scheme-default.css:29]: Right! I added it to the import

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css (hunk 2 above)
Comment (baptisten) [reply src/stories/Atoms/GlobalBranding/GlobalBranding.css]: I corrected it

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css (hunk 3 above)
Comment (baptisten) [reply src/stories/Atoms/GlobalBranding/GlobalBranding.css]: I understand... I checked how things were done on TC.
But this comment makes sense.

File: src/stories/Atoms/GlobalBranding/GlobalBranding.css (hunk 4 above)
Comment (baptisten) [reply src/stories/Atoms/GlobalBranding/GlobalBranding.css]: I corrected it.
I didn't know this syntax, so thanks for sharing the info!

**Review by sleepingmonk — APPROVED** [review]
